### PR TITLE
corrected wrong inheritance in matrixfreeoperator

### PR DESCRIPTION
**Pull request by JensWehner**
matrixfreeoperator inherited from the wrong type, I have no idea why it worked.

### CODE REVIEW

File: include/votca/xtp/matrixfreeoperator.h [hunk 1]
```diff
@@ -23,8 +23,26 @@
...
+}  // namespace Eigen
+
+namespace votca {
+namespace xtp {
+
+class MatrixFreeOperator : public Eigen::EigenBase<MatrixFreeOperator> {
  public:
+  typedef double Scalar;
+  typedef double RealScalar;
+  typedef int StorageIndex;
```
Comment (NicoRenaud) [inline include/votca/xtp/matrixfreeoperator.h]: you made me remove that not so long ago

Comment (NicoRenaud) [reply include/votca/xtp/matrixfreeoperator.h]: do we need them now ?

File: include/votca/xtp/matrixfreeoperator.h [hunk 2]
```diff
@@ -23,8 +23,26 @@
 namespace votca {
 namespace xtp {
 
-class MatrixFreeOperator : public Eigen::EigenBase<Eigen::MatrixXd> {
+class MatrixFreeOperator;
+}
+}  // namespace votca
+namespace Eigen {
+namespace internal {
+// MatrixReplacement looks-like a SparseMatrix, so let's inherits its traits:
```
Comment (NicoRenaud) [inline include/votca/xtp/matrixfreeoperator.h]: can you remove that comment or at least make it correct ?
It's not a SparseMatrix

File: include/votca/xtp/matrixfreeoperator.h [hunk 3]
```diff
@@ -23,8 +23,26 @@
...
+template <>
+struct traits<votca::xtp::MatrixFreeOperator>
+    : public Eigen::internal::traits<Eigen::MatrixXd> {};
+}  // namespace internal
+}  // namespace Eigen
+
+namespace votca {
+namespace xtp {
+
+class MatrixFreeOperator : public Eigen::EigenBase<MatrixFreeOperator> {
```
Comment (NicoRenaud) [inline include/votca/xtp/matrixfreeoperator.h]: The change is a bit above my pay grade I'm afraid. But the eigen web page agrees with you.

File: include/votca/xtp/matrixfreeoperator.h (hunk 1 above)
Comment (JensWehner) [reply include/votca/xtp/matrixfreeoperator.h]: yes

File: include/votca/xtp/matrixfreeoperator.h (hunk 3 above)
Comment (JensWehner) [reply include/votca/xtp/matrixfreeoperator.h]: It is called the Curiously recurring template pattern, basically it is a compile time inheritance MatrixFreeOperator can use the methods of Eigen::Base

File: include/votca/xtp/matrixfreeoperator.h (hunk 1 above)
Comment (JensWehner) [reply include/votca/xtp/matrixfreeoperator.h]: nope

**Pull request merged**